File: frontend/src/types/documents.ts
export interface DocumentListItem {
  id: string;
  source_type: string;
  source_id: string;
  title: string;
  url: string | null;
  author: string | null;
  tags_json: string[];
  acl_policy_id: string | null;
  current_chunk_version: number;
  indexed_at: string | null;
  index_status: string;
  index_error: string | null;
  index_attempts: number;
  index_requested_at: string | null;
  created_at: string;
  updated_at: string;
  chunk_count: number;
}

export interface DocumentChunkRead {
  id: string;
  chunk_index: number;
  chunk_version: number;
  content: string;
  token_count: number;
  acl_policy_id: string | null;
  metadata_json: Record<string, unknown>;
}

export interface DocumentDetail {
  id: string;
  tenant_id: string;
  source_type: string;
  source_id: string;
  url: string | null;
  title: string;
  author: string | null;
  source_created_at: string | null;
  source_updated_at: string | null;
  tags_json: string[];
  metadata_json: Record<string, unknown>;
  acl_policy_id: string | null;
  current_chunk_version: number;
  indexed_at: string | null;
  index_status: string;
  index_error: string | null;
  index_attempts: number;
  index_requested_at: string | null;
  created_at: string;
  updated_at: string;
  chunks: DocumentChunkRead[];
}

export interface ReindexResponse {
  indexed_documents: number;
  indexed_chunks: number;
}

export interface ChunkSearchResultItem {
  document_id: string;
  document_title: string;
  document_url: string | null;
  source_type: string;
  chunk_id: string;
  chunk_index: number;
  snippet: string;
  score: number;
  ranker: string;
}

File: backend/app/schemas/documents.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class DocumentChunkRead(BaseModel):
    id: str
    chunk_index: int
    chunk_version: int
    content: str
    token_count: int
    acl_policy_id: str | None = None
    metadata_json: dict[str, Any]


class DocumentListItem(BaseModel):
    id: str
    source_type: str
    source_id: str
    title: str
    url: str | None = None
    author: str | None = None
    tags_json: list[str]
    acl_policy_id: str | None = None
    current_chunk_version: int
    indexed_at: datetime | None = None
    index_status: str
    index_error: str | None = None
    index_attempts: int
    index_requested_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    chunk_count: int


class DocumentDetail(BaseModel):
    id: str
    tenant_id: str
    source_type: str
    source_id: str
    url: str | None = None
    title: str
    author: str | None = None
    source_created_at: datetime | None = None
    source_updated_at: datetime | None = None
    tags_json: list[str]
    metadata_json: dict[str, Any]
    acl_policy_id: str | None = None
    current_chunk_version: int
    indexed_at: datetime | None = None
    index_status: str
    index_error: str | None = None
    index_attempts: int
    index_requested_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    chunks: list[DocumentChunkRead]


class ReindexRequest(BaseModel):
    document_ids: list[str] = Field(default_factory=list)
    source_type: str | None = None
    acl_policy_id: str | None = None


class ReindexResponse(BaseModel):
    indexed_documents: int
    indexed_chunks: int


class ResetEmbeddingsResponse(BaseModel):
    reset_documents: int
    deleted_chunks: int
    status: str


class ForgetDocumentResponse(BaseModel):
    status: str
    document_id: str


class ChunkSearchRequest(BaseModel):
    query: str = Field(min_length=2, max_length=300)
    limit: int = Field(default=8, ge=1, le=50)


class ChunkSearchResultItem(BaseModel):
    document_id: str
    document_title: str
    document_url: str | None = None
    source_type: str
    chunk_id: str
    chunk_index: int
    snippet: str
    score: float
    ranker: str


class ChunkSearchResponse(BaseModel):
    results: list[ChunkSearchResultItem]
